testing/src/main/java/org/oppia/android/testing/lightweightcheckpointing/ExplorationCheckpointTestHelper.kt

```diff
@@ -49,36 +63,126 @@ class ExplorationCheckpointTestHelper @Inject constructor(
   lateinit var explorationCheckpointCaptor: ArgumentCaptor<AsyncResult<ExplorationCheckpoint>>
 
   /**
-   *  Saves a fake checkpoint for explorationId [FAKE_EXPLORATION_ID_1] for the specified profileId.
-   *  The size of the checkpoint saved here is 67 bytes.
+   * Saves a checkpoint for topic Fractions, story 0, exploration 0.
+   *
+   * @param profileId the profileID for which the checkpoint has to be saved
+   * @param version the version of the exploration for which the checkpoint has to be created
    */
-  fun saveFakeExplorationCheckpoint(internalProfileId: Int) {
-    val checkpoint = createFakeExplorationCheckpoint(FAKE_EXPLORATION_TITLE_1, timestamp = 0L)
+  fun saveCheckpointForFractionsStory0Exploration0(profileId: ProfileId, version: Int) {
+    val checkpoint = createExplorationCheckpoint(
+      explorationTitle = FRACTIONS_EXPLORATION_0_TITLE,
+      pendingStateName = FRACTIONS_STORY_0_EXPLORATION_0_FIRST_STATE_NAME,
+      version = version
+    )
+    val saveCheckpointDataProvider = explorationCheckpointController.recordExplorationCheckpoint(
+      profileId,
+      FRACTIONS_EXPLORATION_ID_0,
+      checkpoint
+    )
+    verifyProviderFinishesWithSuccess(saveCheckpointDataProvider)
+  }
+
+  /**
+   * Saves a checkpoint for topic Fractions, story 0, exploration 1.
+   *
+   * @param profileId the profileID for which the checkpoint has to be saved
+   * @param version the version of the exploration for which the checkpoint has to be created
+   */
+  fun saveCheckpointForFractionsStory0Exploration1(profileId: ProfileId, version: Int) {
+    val checkpoint = createExplorationCheckpoint(
+      explorationTitle = FRACTIONS_EXPLORATION_1_TITLE,
+      pendingStateName = FRACTIONS_STORY_0_EXPLORATION_1_FIRST_STATE_NAME,
+      version = version
+    )
+    val saveCheckpointDataProvider = explorationCheckpointController.recordExplorationCheckpoint(
+      profileId,
+      FRACTIONS_EXPLORATION_ID_1,
+      checkpoint
+    )
+    verifyProviderFinishesWithSuccess(saveCheckpointDataProvider)
+  }
+
+  /**
+   * Updates the saved checkpoint for Fractions, story 0, exploration 0. For this function to work
+   * correctly it should be called after [saveCheckpointForFractionsStory0Exploration0].
+   *
+   * @param profileId the profileID for which the checkpoint has to be saved
```

Comment:
```suggestion
   * @param profileId the profile ID for which the checkpoint has to be saved
```

Ditto elsewhere in the PR.

Reply:
done